sql-common/client.c

```diff
@@ -1897,15 +1897,27 @@ mysql_get_ssl_cipher(MYSQL *mysql __attribute__((unused)))
 
 #if defined(HAVE_OPENSSL) && !defined(EMBEDDED_LIBRARY)
 
+#include <openssl/x509v3.h>
+
+#if !defined(HAVE_YASSL) && !defined(LIBRESSL_VERSION_NUMBER)
+  #if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10002000L
+    #define HAVE_X509_check_host 1
```

Comment:
The problem is that `sql_common/client.c` is used in 3 different  targets (`libmysql`, `libmysqld` and `mysqld`). In my opinion, it's better to have these defines in one place rather than put `CHECK_FUNCTION_EXISTS` in 3 different `CMakeLists.txt` files and have potential  "updated only in one place" problem. Leaving as is.